app/components/profile.jsx

```diff
@@ -19,73 +23,60 @@ export class Profile extends Component {
     this.reset();
   }
 
-  componentWillUpdate (props) {
-    this.reset(props);
+  componentWillUpdate ({ params: { username } }) {
+    if (username !== this.props.params.username) {
+      this.reset();
+    }
   }
 
-  reset (props) {
-    const { checkVersion } = props || this.props;
+  reset () {
```

Comment:
since we're sometimes passing in new props into this function we need to be able to utilize it, so we'll be listing `props` as an argument. if `this.reset` is called without any arguments (e.g. `this.reset()` vs `this.reset(nextProps)`, `props` will be `undefined`, and we'll be utilizing this fact in a bit.

final result:

```js
reset (props) {
```

you should also do this in `tracker.jsx` (line 34)